Code/Framework/AzCore/AzCore/Serialization/Json/JsonImporter.cpp

```diff
@@ -144,21 +160,606 @@ namespace AZ
+                mergePatchResult.Combine(ResolveImportsInOrder(*newFieldValue, allocator, fieldValue,
+                    importPathStack, settings, element));
+            }
+        }
+        else if (fieldValue.IsArray())
+        {
+            rapidjson::Value* newFieldValue{};
+            if (source.IsObject())
+            {
+                rapidjson::Value name;
+                name.CopyFrom(rapidjson::Value(fieldNameStringRef), allocator, true);
+                // Add an empty object or array field depending on the field value type
+                target.AddMember(AZStd::move(name),
+                    rapidjson::Value{ rapidjson::kArrayType },
+                    allocator);
+                auto newFieldMemberIter = target.FindMember(fieldNameStringRef);
+                if (newFieldMemberIter == target.MemberEnd())
+                {
+                    AZStd::string_view fieldName(static_cast<const char*>(fieldNameStringRef),
+                        fieldNameStringRef.length);
+                    auto reportMessage = ReporterString::format("Failed to add member %.*s to JSON while importing.",
+                        AZ_STRING_ARG(fieldName));
+                    if (!importPathStack.empty())
+                    {
+                        reportMessage += ReporterString::format(R"( This occured while importing file "%s".)",
+                            importPathStack.back().c_str());
+                    }
+                    return settings.m_reporting(reportMessage, ResultCode(Tasks::Import, Outcomes::Catastrophic), element);
+                }
+
+                // Get reference to the value object/array member that was just added
+                newFieldValue = &newFieldMemberIter->value;
+            }
+            else
+            {
+                // The source value is an array in this if block
+                rapidjson::SizeType oldArrayCount = target.Size();
+                target.PushBack(rapidjson::Value{ rapidjson::kArrayType }, allocator);
+
+                // Verify that a new array element was added
+                if (oldArrayCount == target.Size())
+                {
+                    ReporterString reportMessage{ "Failed to Add array element to JSON while importing." };
+                    if (!importPathStack.empty())
+                    {
+                        reportMessage += ReporterString::format(R"( This occured while importing file "%s".)",
+                            importPathStack.back().c_str());
+                    }
+                    return settings.m_reporting(reportMessage, ResultCode(Tasks::Import, Outcomes::Catastrophic), element);
+                }
+
+                newFieldValue = &target[oldArrayCount];
+            }
+
+            // Array fields are recursively checked for children $import directives
+            mergePatchResult.Combine(ResolveImportsInOrder(*newFieldValue, allocator, fieldValue,
+                importPathStack, settings, element));
```

Comment:
This all looks cut-n-pasted from fieldValue.IsObject() case directly above. Can it be turned into a helper function (or even internal lambda) to make it shared logic? It would be extremely hard to spot if the two ever got slightly out of sync accidentally.

Reply:
Yeah, the logic is similar with some changes in the middle to make a rapidjson::Value that is an Array instead of an Object.